pkg/cloudprovider/cloud.go

```diff
@@ -74,8 +76,10 @@ type TCPLoadBalancer interface {
 type Instances interface {
 	// NodeAddresses returns the addresses of the specified instance.
 	NodeAddresses(name string) ([]api.NodeAddress, error)
-	// ExternalID returns the cloud provider ID of the specified instance.
+	// ExternalID returns the cloud provider ID of the specified instance (obsolete).
 	ExternalID(name string) (string, error)
+	// InstanceID returns the cloud provider ID of the specified instance.
+	GetInstanceID(name string) (string, error)
```

Comment:
I meant in the Kubernetes API. In cloudprovider.Interface it's a bit redundant. InstanceID would be ok.